### ISSUE TITLE
Playback for file returns to series main screen from "retrieving"

### ISSUE THREAD

**Issue by jcdick1** [jellyfin-archive/jellyfin-roku-legacy#134]
**Describe the bug**
When selecting an unwatched item in a series, the "retrieving ..." progress graph comes on screen, pauses at about 40%, then disappears and the screen is returned to the title main screen, instead of the season screen from which the file was chosen.

**To Reproduce**

1. Go to 'Any series'
2. Click on 'Any season with unwatched episodes'
3. Scroll down to 'Any unwatched episode'
4. Select to play

**Expected behavior**
File plays

**Logs**
03-01 02:55:48.156 [beacon.signal] |AppLaunchInitiate ---------> TimeBase(0)
03-01 02:55:48.156 [beacon.signal] |AppCompileInitiate --------> TimeBase(0 ms)
03-01 02:55:48.156 [scrpt.cmpl] Compiling 'Jellyfin', id 'dev'
03-01 02:55:48.167 [scrpt.load.mkup] Loading markup dev 'Jellyfin'
03-01 02:55:48.174 [scrpt.unload.mkup] Unloading markup dev 'Jellyfin'
03-01 02:55:48.218 [scrpt.parse.mkup.time] Parsed markup dev 'Jellyfin' in 51 mi                                                     lliseconds

------ Compiling dev 'Jellyfin' ------
03-01 02:55:48.271 [scrpt.ctx.cmpl.time] Compiled 'Jellyfin', id 'dev' in 52 mil                                                     liseconds
03-01 02:55:48.460 [scrpt.proc.mkup.time] Processed markup dev 'Jellyfin' in 182                                                      milliseconds
03-01 02:55:48.461 [beacon.signal] |AppCompileComplete --------> Duration(305 ms                                                     ), 238.95 KiP
03-01 02:55:48.470 [ui.frm.plugin.running.enter] Entering PLUGIN_RUNNING for dev
03-01 02:55:48.471 [beacon.signal] |AppLaunchInitiate ---------> TimeBase(0)
03-01 02:55:50.023 [scrpt.ctx.run.enter] UI: Entering 'Jellyfin', id 'dev'

------ Running dev 'Jellyfin' main ------
Get server details
[RALE][INFO] - Roku Advanced Layout Editor v.2.0.9
[RALE][INFO] - TrackerTask run
[RALE][INFO] - Waiting on ECP input request
HI
Get user login
[RALE][INFO] - Roku Advanced Layout Editor v.2.0.9
[RALE][INFO] - TrackerTask run
[RALE][INFO] - Waiting on ECP input request
HI
HI
[RALE][INFO] - Roku Advanced Layout Editor v.2.0.9
[RALE][INFO] - TrackerTask run
[RALE][INFO] - Waiting on ECP input request
BRIGHTSCRIPT: ERROR: roSGNode.AddReplace: "currentpage": Type mismatch: pkg:/com                                                     ponents/Pager.brs(79)
roSGNode
roSGNode
roSGNode
03-01 03:07:36.788 [beacon.signal] |AppExitInitiate -----------> TimeBase(708303                                                      ms)
03-01 03:07:36.802 [beacon.signal] |AppExitComplete -----------> Duration(28 ms)
03-01 03:07:36.803 [beacon.header] __________________________________________
03-01 03:07:36.803 [beacon.report] |AppLaunchInitiate ---------> TimeBase(0)
03-01 03:07:36.803 [beacon.report] |AppExitInitiate -----------> TimeBase(708303                                                      ms)
03-01 03:07:36.803 [beacon.report] |AppExitComplete -----------> Duration(28 ms)
03-01 03:07:36.803 [beacon.footer] __________________________________________

**Screenshots**

**Additional context**
Roku 4620X ("Premiere") v9.2.0

**Comment by cewert** [comment]
Have you verified the file isn't corrupt? Does it play on your PC or another client? I've had a corrupt file play fine and then kick me back out half way through

**Comment by jcdick1** [comment]
Yes, I can play it via the web client and Android client, on LAN and remote.  And I tried multiple shows and seasons.  I can try to upload the almost 2 minute video I took showing the behavior.

**Comment by bisby** [comment]
`roSGNode.AddReplace: "currentpage": Type mismatch: pkg:/com ponents/Pager.brs(79)`

points to

https://github.com/jellyfin/jellyfin-roku/blob/master/components/Pager.brs#L79

Which is trying to set a none integer value into currentpage.

It seems like the episode page is still using a paging handler, even though it got changed to a vertical scroll list.

**Comment by cewert** [comment]
I think he could have been on any page when he got the error. It doesn't seem like the error crashed the app since `roSGNode` and `[beacon.signal]` come after, meaning he got the error before he played the video.

I'm curious why `roSGNode` shows up 3 times in the log before the video plays? When I play a video it only shows up once

**Comment by cewert** [comment]
I did a quick test and I could only get `roSGNode` to show up 3 times by doing one of the following:

- Press "ok" 3 times really fast when selecting the episode. This spawned 3 different videos on top of each other. pressing back went to the previous one until it ran out of videos and crashed.
- Press "ok" once on an episode then quickly before it finishes loading hitting back. This logged `roSGNode` with no `[beacon.signal]`. Repeat 2 times, then actually let it play the 3rd time.

I didn't let a video play to 40% to see if it would kick me back out but the episode did play both times.

**Comment by thomabx** [comment]
The behavior sounds like an issue with transcoding. Either a bad request from the app or an unhandled error from the server.

It sounds like the server is responding with a transcode playlist (which may or may not be empty) and after some time enters the state of "finished" and also that the current position has met or passed the length of the video. Both states are caught by the main loop which causes it to send it back a screen twice. There is no check to prevent this but I think the primary issue is that transcoding doesn't work.

@jcdick1, does this affect all videos or specific videos?

**Comment by thomabx** [comment]
> I didn't let a video play to 40% to see if it would kick me back out but the episode did play both times.

I believe @jcdick1 means that the loading bar gets to 40%, not the actual playback. Before transcoding was added, the loading bar would go to ~40% (my Roku does not list percentages but that sounds right) and nothing would happen.

**Comment by cewert** [comment]
@thomabx Yea you're right I misread that

The log shows he was setting up the server and user settings. Maybe that explains the pager bug? The two "HI"s before the error are from opening the settings dialog.

**Comment by jcdick1** [comment]
I resolved the issue, as there was a transcoding error, as @thomabx thought.  So this can be closed, unless you want to flag it for the brightscript error in Pager.brs

Also, would it be any help if I put in tickets when I get warnings and errors in the telnet output that I don't see tickets for, or should I assume you've already seen them in your own testing?

**Comment by bisby** [comment]
Thanks for the feedback and giving us things to keep an eye on. We'll re-open a new issue if that pager thing comes back to bite us.
